Use poise-python cookbook

gsreynolds opened the pull request:
Hi! Noticed #43 and just wanted to contribute poise-python based duplicity install we already had in our duplicity wrapper cookbook. Hope it helps!

chr4 commented on attributes/default.rb:
```diff
@@ -33,13 +33,13 @@
 default['duplicity_ng']['source']['gnupg']['version'] = '0.3.2'
 default['duplicity_ng']['source']['gnupg']['url'] = "http://downloads.sourceforge.net/project/py-gnupg/GnuPGInterface/#{node['duplicity_ng']['source']['gnupg']['version']}/GnuPGInterface-#{node['duplicity_ng']['source']['gnupg']['version']}.tar.gz" # rubocop:disable Metrics/LineLength
-default['duplicity_ng']['source']['gnupg']['checksum'] = '0ea672251e2e3f71b62fef0c01539519d500f6b338f803af6b57e67a73cca8e6'
+# default['duplicity_ng']['source']['gnupg']['checksum'] = '0ea672251e2e3f71b62fef0c01539519d500f6b338f803af6b57e67a73cca8e6'
```
Please remove the commented out checksums

chr4 commented on recipes/install_azure.rb:
```diff
@@ -20,12 +20,14 @@
 
 # Install Azure-specific packages
-node['duplicity_ng']['source']['azure']['packages'].each { |n| package n }
+package node['duplicity_ng']['source']['azure']['packages'] do
```
Does Chef now support `package [array]`?

chr4 commented on recipes/source.rb:
```diff
@@ -22,48 +22,48 @@
 else
-  node['duplicity_ng']['source']['python']['packages'].each do |name|
-    package name
+  package node['duplicity_ng']['source']['python']['packages'] do
```
See `package [array]` question above

rmoriz commented:
Does the combination of package + pip addons work on centos?

gsreynolds commented:
Thanks, happy to help out. I'll remove the comments and verify pip add-ons
tomorrow.

Chef has supported multipackage installs since 12.1 which was released 3
years ago, I thought it might be safe enough to swap over to the new
syntax. See https://docs.chef.io/resource_package.html#multiple-packages
and
https://blog.chef.io/2015/03/03/phil-dibowitz-chef-client-12-1-0-multipackage-installs/.
If you prefer, I think either the compat_resource or multipackage cookbooks
provide compat back to to 12.0.
On Thu, 20 Apr 2017 at 18:00, Roland Moriz <notifications@github.com> wrote:

> Does the combination of package + pip addons work on centos?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/chr4-cookbooks/duplicity_ng/pull/45#issuecomment-295814618>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdmeriLZx5TQeK4IUucItInB_HIjLORks5rx48ggaJpZM4NDO3L>
> .
>

gsreynolds commented:
Hi, updated the PR now to address the requested changes. Verified as working under package + pip addons, source + pip addons and pip + pip addons for centos and debian.

Also added an install_par2 recipe for the Par2 Wrapper Backend which we are using `par2+scheme://[user[:password]@]host[:port]/[/]path`

The only issues I can see, not related to this PR, are that the `python-gnupginterface` package doesn't exist in debian > 7. (https://packages.debian.org/search?keywords=python-gnupginterface) and `python-swiftclient`can't be found under centos 7.

```
* No candidate version available for python-swiftclient
           ================================================================================
           Error executing action `install` on resource 'yum_package[python-swiftclient]'
           ================================================================================
```

gsreynolds commented:
> Does the combination of package + pip addons work on centos?
Oops, just realised what you meant. Pushed a commit with another suite package-pip to cover this combination.

chr4 commented on .kitchen.yml:
```diff
@@ -33,22 +31,54 @@ suites:
+    - recipe[duplicity_ng::install_swift]
     attributes:
       duplicity_ng:
-        path: /usr/bin/duplicity
```
Was there a reason to switch from the fully-qualified path to `duplicity`?

gsreynolds replied:
Yes, cross platform issues I mentioned in 72090c9 "... Fix duplicity paths under source suites, can either be /usr/bin/local/duplicity under debian family or /usr/bin/duplicity under rhel-ish." 

Either way, regardless of install method or platform, you would expect it to end up in the path?

chr4 replied:
Makes sense.

gsreynolds commented:
Thanks for approving @chr4 :+1:

chr4 commented:
LGTM, merging. Thanks for the contribution! Mind being added to the contributors list in the README?

gsreynolds commented:
Don't mind at all, happy for my name and email on my profile to be listed.

chr4 commented:
Ok. Added you to the list of contributors, and released `2.0.0` on the Supermarket. I've chosen a major release for this.
Thanks again!
